sync-core/src/main/java/com/cloudant/sync/replication/BasicPushStrategy.java

```diff
@@ -280,13 +281,13 @@ private int processOneChangesBatch(Changes changes) throws AttachmentException,
 
     private ItemsToPush missingRevisionsToJsonDocs(
             Map<String, DocumentRevisionTree> allTrees,
-            Map<String, Set<String>> revisions) throws AttachmentException {
+            Map<String, CouchClient.MissingRevisions> revisions) throws AttachmentException {
```

Comment:
I had a real pain figuring and keeping track of what was local and what was remote state in this method. Perhaps rename some things as part of the work in this method? `allLocalTrees` and `missingRemoteRevisions` maybe to start with? If I've even got that right...